calico/reference/felix/configuration.md

```diff
@@ -51,7 +51,7 @@ The full list of parameters which can be set is as follows.
 | `HealthEnabled`                   | `FELIX_HEALTHENABLED`                   | When enabled, exposes felix health information via an http endpoint. | boolean |
 | `HealthHost`                      | `FELIX_HEALTHHOST`                      | The address on which Felix will respond to health requests. [Default: `localhost`] | string |
 | `HealthPort`                      | `FELIX_HEALTHPORT`                      | The port on which Felix will respond to health requests. [Default: `9099`] | int |
-| `IpInIpEnabled`                   | `FELIX_IPINIPENABLED`                   | Whether Felix should configure an IPinIP interface on the host. Set automatically to `true` by `{{site.nodecontainer}}` or `calicoctl` when you create an IPIP-enabled pool. [Default: `false`] | boolean |
+| `DeprecatedIpInIpEnabled`                   | `FELIX_IPINIPENABLED`                   | Optional, you shouldn't need to change this setting as Felix calculates if IPIP should be enabled based on the existing IP Pools. When set, this overrides whether Felix should configure an IPinIP interface on the host. [Default: `nil` (unset)] | \*bool |
```

Comment:
The intention of this was to warn future developers that the field should not be accessed directly, while still maintaining the user facing API, which I think is a worthwhile goal.

It would be rather easy for someone to accidentally reference this field instead of the new aggregate fields and end up with a broken feature.

It was unfortunate that the env var and config file name depends on actual name of the value in the code, for this use case. But from the sounds of it there is no reasonable way to achieve that?

Reply:
I had thought that by leaving the FelixConfiguration and env var unchanged we could change the variable name in config_params, but I guess I overlooked how it's loaded from a config file or env var. I'll remove the "deprecated" prefix from the variables. Thanks @fasaxc !